docs/docs/adrs/adr-009-soft-opt-out.md

```diff
@@ -0,0 +1,48 @@
+---
+sidebar_position: 10
+title: Soft Opt-Out
+---
+## ADR 009: Soft Opt-Out
+
+## Changelog
+
+* 6/13/23: Initial draft of ADR. Feature already implemented and in production.
+
+## Status
+
+Accepted
+
+## Context
+
+Some small validators may not have the resources needed to validate all consumer chains. Therefore a need exists to allow the bottom _% of validators to opt-out of validating a consumer chain. Meaning downtime infractions for relevant validators are dropped without ever reaching the provider.
+
+This document specifies a modification to the ccv protocol which allows the bottom x% of the validator set by power to opt out of validating consumer chains without being jailed or otherwise punished for it. The feature is implemented with entirely consumer-side code.
+
+## Decision
+
+A consumer param exists, known as `SoftOptOutThreshold`, which is a string decimal in the range of [0, 0.2], that determines the portion of validators which are allowed to opt out of validating that specific consumer.
+
+Every consumer beginblocker, a function is ran which determines the voting power s.t. any validator with greater than or equal to said voting power must validate that specific consumer. Any validator with less than said voting power may opt out of validating that consumer chain.
+
+The voting power value described above is referred to in code as "smallest non opt out power". The value is recomputed every beginblocker in `UpdateSmallestNonOptOutPower()`. In a nutshell, the method obtains the total voting power of the consumer, iterates through the full valset (ordered power ascending) keeping track of a power sum, and when `powerSum / totalPower > SoftOptOutThreshold`, the `SmallestNonOptOutPower` is found and persisted.
+
+Then, whenever the `Slash()` interface is executed on the consumer, if the voting power of the relevant validator being slashed is less than `SmallestNonOptOutPower` for that block, the slash request is dropped and never sent to the provider.
+
+## Consequences
+
+### Positive
+
+* Small validators can opt out of validating specific consumers without being punished for it.
+
+### Negative
+
+* Small validators who are not being slashed will still be present in the validator set of the consumer chain. In other words, we are sacrificing liveness, not safety. The economic security of the consumer chain is the same, but consumers may be more likely to halt. If soft opt out threshold is set to 10% for example, and every validator who doesn't have to validate the consumer doesn't validate it, then 23% downtime of the remaining valset could halt the chain. This may manifest in slightly longer downtime periods during upgrades.
```

Comment:
```suggestion
The bottom `x%` is still part of the total voting power of the consumer chain. This means that if the soft opt-out threshold is set to `10%` for example, and every validator in the bottom `10%` opts out from validating the consumer, then a `24%` downtime of the remaining voting power would halt the chain. This may be especially problematic during consumer upgrades.
```